### PR TITLE
[SofaBase] Package SofaBaseCollision

### PR DESCRIPTION
**Pull request by fredroy**
#1633

- Usual cleaning (pragma, namespaces)
- remove/reorganize useless includes
- remove MechanicalObject header-dependency (not reflecting with library linking)
- use only RGBAColor
- remove warnings

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/CubeModel.cpp [hunk 1]
```diff
@@ -184,12 +176,7 @@ void CubeCollisionModel::draw(const core::visual::VisualParams* vparams)
         ++level;
         color *= 0.8f;
     }
-    Vec<4,float> c;
-    if (isSimulated())
-        c=Vec<4,float>(1.0f, 1.0f, 1.0f, color);
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/CubeModel.cpp:189]: if it is like TriangleModel, I think simulated is "gold" and not is white

File: SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/CylinderModel.cpp [hunk 2]
```diff
@@ -42,13 +36,7 @@ int RigidCylinderCollisionModelClass = core::RegisterObject("Collision model whi
         .addAlias("CylinderModel")
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/CylinderModel.cpp:36]: didn't even know we had that in SOFA ....

File: SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/NewProximityIntersection.h [hunk 3]
```diff
@@ -52,20 +45,12 @@ class SOFA_BASE_COLLISION_API NewProximityIntersection : public BaseProximityInt
 
 };
 
-} // namespace collision
+} // namespace sofa::core::collision
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/NewProximityIntersection.h]: sofa::component::collision no sofa::core

yes... I'm really reading!

File: SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/config.h.in [hunk 4]
```diff
@@ -19,16 +19,16 @@
 *                                                                             *
 * Contact information: contact@sofa-framework.org                             *
 ******************************************************************************/
-#ifndef SOFABASECOLLISION_CONFIG_H
-#define SOFABASECOLLISION_CONFIG_H
+#pragma once
 
-#include <SofaBase/config.h>
+#include <sofa/config.h>
+#include <sofa/config/sharedlibrary_defines.h>
```
Comment (epernod) [inline SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/config.h.in:25]: what is this header?

Comment (fredroy) [reply SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/config.h.in:25]: This one ?
`
#include <sofa/config/sharedlibrary_defines.h>
`
It defines the macros SOFA_EXPORT_DYNAMIC_LIBRARY and SOFA_IMPORT_DYNAMIC_LIBRARY

File: SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/CubeModel.cpp (hunk 1 above)
Comment (fredroy) [reply SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/CubeModel.cpp:189]: I was not sure but I went for the lazy way :p

File: SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/CylinderModel.cpp (hunk 2 above)
Comment (fredroy) [reply SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/CylinderModel.cpp:36]: 😙

File: SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/NewProximityIntersection.h (hunk 3 above)
Comment (fredroy) [reply SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/NewProximityIntersection.h]: right.
(And i dont doubt about it, with your remarks on end of line or spaces 😙)

**Pull request merged**